src/validation.cpp

```diff
@@ -2492,20 +2502,36 @@ bool IsValidPeginWitness(const CScriptWitness& pegin_witness, const COutPoint& p
 
     uint256 block_hash;
     uint256 tx_hash;
-
     // Get txout proof
-    Sidechain::Bitcoin::CMerkleBlock merkle_block;
-    if (!GetBlockAndTxFromMerkleBlock(block_hash, tx_hash, merkle_block, stack[5])) {
-        return false;
-    }
-    if (!CheckBitcoinProof(block_hash, merkle_block.header.nBits)) {
-        return false;
-    }
+    if (Params().GetConsensus().parent_chain_signblockscript == CScript()) {
 
-    // Get serialized transaction
-    Sidechain::Bitcoin::CTransactionRef pegtx;
-    if (!CheckPeginTx(stack[4], pegtx, prevout, value, claim_script)) {
-        return false;
+        Sidechain::Bitcoin::CMerkleBlock merkle_block_pow;
+        if (!GetBlockAndTxFromMerkleBlock(block_hash, tx_hash, merkle_block_pow, stack[5])) {
+            return false;
+        }
+        if (!CheckBitcoinProof(block_hash, merkle_block_pow.header.nBits)) {
```

Comment:
Now we have two very similar things with different naming conventions, `CheckBitcoinProof` and `CheckProofParent`.

Can we just abstract this out to `CheckProofParent` with a boolean arg switch?

Reply:
Yes, I just wanted to avoid changing CheckProof since it is called a few times. I could use a boolean that defaults to false, but I tend to dislike default arguments in C++. I find ```CheckProofParent(merkle_block.header, Params().GetConsensus())``` more clear than ```CheckProof(merkle_block.header, Params().GetConsensus(), true)```, which forces me to go to the function and see what the boolean is about. If it was python, ```CheckProof(merkle_block.header, Params().GetConsensus(), use_parent_chain_script=True)``` is fine. Perhaps I can just add a comment when using the boolean.

Reply:
fair enough, perhaps a minor renaming then to make them obviously related

Reply:
or heck just comment what it's doing

Reply:
Renamed to CheckProofSignedParent as discussed on chat.